components/job-orchestration/job_orchestration/scheduler/search/search_scheduler.py

```diff
@@ -114,31 +119,31 @@ def set_job_status(
     return rval
 
 
-async def handle_cancelling_search_jobs(db_conn) -> None:
+async def handle_cancelling_search_jobs(db_conn_pool) -> None:
     global active_jobs
 
-    with contextlib.closing(db_conn.cursor(dictionary=True)) as cursor:
-        cancelling_jobs = fetch_cancelling_search_jobs(cursor)
-        db_conn.commit()
+    with contextlib.closing(db_conn_pool.connect()) as db_conn:
+        cancelling_jobs = fetch_cancelling_search_jobs(db_conn)
 
-    for job in cancelling_jobs:
-        job_id = job["job_id"]
-        if job_id in active_jobs:
-            job = active_jobs.pop(job_id)
-            cancel_job_except_reducer(job)
-            # Perform any async tasks last so that it's easier to reason about synchronization
-            # issues between concurrent tasks
-            await release_reducer_for_job(job)
-        else:
-            continue
-        if set_job_status(
-            db_conn, job_id, SearchJobStatus.CANCELLED, prev_status=SearchJobStatus.CANCELLING
-        ):
-            logger.info(f"Cancelled job {job_id}.")
-        else:
-            logger.error(f"Failed to cancel job {job_id}.")
+        for job in cancelling_jobs:
```

Comment:
Just wonder is there a specific reason you indent the code snippet so it's under `with contextlib.closing(db_conn_pool.connect()) as db_conn:`?

Reply:
Right, the way the sqlalchemy connection pool works is that you call close() on a connection to return it to the pool. The with closing() blocks allow us to guarantee that the connections always get returned to the pool.